File: mobile/lib/models/usuario.dart
class Usuario {
  final int? id;
  final String nome;
  final String email;
  final String telefone;
  final bool motorista;
  final bool emailValidado;
  final String role;
  final String? criadoEm;

  Usuario({
    this.id,
    required this.nome,
    required this.email,
    required this.telefone,
    required this.motorista,
    required this.emailValidado,
    required this.role,
    this.criadoEm,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      id: json['id'] as int?,
      nome: json['nome'] as String? ?? '',
      email: json['email'] as String? ?? '',
      telefone: json['telefone'] as String? ?? '',
      motorista: json['motorista'] as bool? ?? false,
      emailValidado: json['emailValidado'] as bool? ?? false,
      role: json['role'] as String? ?? 'PASSAGEIRO',
      criadoEm: json['criadoEm'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nome': nome,
      'email': email,
      'telefone': telefone,
      'motorista': motorista,
      'emailValidado': emailValidado,
      'role': role,
      'criadoEm': criadoEm,
    };
  }
}

File: ms-usuarios/src/main/java/br/edu/ifsp/carona/usuarios/entity/Usuario.java
package br.edu.ifsp.carona.usuarios.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "usuarios")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nome;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(nullable = false)
    private String senha;

    @Column(nullable = false)
    private String telefone;

    @Builder.Default
    private Boolean motorista = false;

    @Builder.Default
    private Boolean emailValidado = false;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private Role role = Role.PASSAGEIRO;

    @CreationTimestamp
    private LocalDateTime criadoEm;

    @UpdateTimestamp
    private LocalDateTime atualizadoEm;
}
